### PR TITLE
S70.56 Remake

### PR DESCRIPTION
**Pull request by pvaass**
The initial setup for this was way too complicated for what it was.

Probably better to just check the files instead of the changes.

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -20,7 +18,7 @@ class Client
     /**
      * @var string
      */
-    protected $base_url = "https://app.klipfolio.com/api";
+    protected $base_url = "https://sandbox.klipfolio.com/api";
```
Comment (swordbeta) [inline src/Client.php]: Hardcoderino

File: src/Object/BaseApiCollection.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+
+namespace ExpandOnline\KlipfolioApi\Object;
+
+
```
Comment (swordbeta) [inline src/Object/BaseApiCollection.php]: Whitespaces...whitespaces everywhere

**Comment by ronoudgenoeg** [comment]
My main issue with this setup is that the request object and the response object are the same.

When I do an API request, I don't expect the response object to have methods like 'getEndpoint', or 'setFilters' (doesn't exist yet, but I imagine this will be needed sometime soon). Especially when getting a collection, you will have an iterator containing objects which all have information about endpoints.

File: src/Connector/BaseApiConnector.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    protected $apiParams = [];
+
+    /**
+     * @param null $option
+     * @return mixed
+     */
+    abstract public function getEndpoint($option = null);
+
+
+    abstract protected function getObjectName();
```
Comment (ronoudgenoeg) [inline src/Connector/BaseApiConnector.php]:
```
protected function getObjectName() {
    return get_class($this);
}
```

File: src/Connector/BaseApiResourceConnector.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
+<?php namespace ExpandOnline\KlipfolioApi\Connector;
+
+use ExpandOnline\KlipfolioApi\Exception\KlipfolioApiException;
+use ExpandOnline\KlipfolioApi\Object\BaseApiResource;
+
+abstract class BaseApiResourceConnector extends BaseApiConnector
+{
+    protected $id = null;
+    protected $resource = null;
```
Comment (ronoudgenoeg) [inline src/Connector/BaseApiResourceConnector.php]: Isn't id simply a part of a resource? Why are these seperate?

File: src/Connector/BaseApiConnector.php (hunk 3 above)
Comment (ronoudgenoeg) [inline src/Connector/BaseApiConnector.php]: Maybe rename this to getResourceObjectName

File: src/Connector/BaseApiResourceConnector.php (hunk 4 above)
Comment (pvaass) [inline src/Connector/BaseApiResourceConnector.php]: @ronoudgenoeg Need to set $id for KlipfolioApi->get

**Pull request merged**